### ISSUE TITLE
Modifying state during view update, this will cause undefined behavior.

### ISSUE THREAD

**Issue by plivesey** [liscio/CollectionThing#4]
I got this runtime warning while running the code. I got this for one of my other preference changes. The only way I could get rid of this warning was to to add a `DisapatchQueue.main.async` which I know you don't want to do.

It seems to occur whenever the view changes because of an observed object changing. I haven't tried to reproduce this yet.

**Comment by plivesey** [comment]
<img width="477" alt="Screen Shot 2019-11-21 at 10 30 08 AM" src="https://user-images.githubusercontent.com/1248589/69365953-3389bf00-0c4a-11ea-810c-2f6168e8c5ea.png">

**Comment by itrack** [comment]
Try with:
DispatchQueue.main.async {
// code to execute
}

**Comment by plivesey** [comment]
Yep, that's what I'm doing in my code base. It's just that @liscio seemed to want to avoid using any dispatches (`No weird uses of DispatchQueue.async` - README)
